flow/tracing.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package flow
+
+import (
+	"encoding/hex"
+	"math/rand"
+
+	flowpb "github.com/cilium/cilium/api/v1/flow"
+)
+
+const traceIDLen = 16
+
+type traceContextOptions struct {
+	traceIDs []string
+}
+
+// TraceContextOption is an option to use with TraceContext.
+type TraceContextOption interface {
+	apply(o *traceContextOptions)
+}
+
+type funcTraceContextOption func(*traceContextOptions)
+
+func (ftco funcTraceContextOption) apply(tco *traceContextOptions) {
+	ftco(tco)
+}
+
+// WithTraceIDs defines a set of trace IDs to select from when generating trace
+// contexts.
+func WithTraceIDs(traceIDs []string) TraceContextOption {
+	return funcTraceContextOption(func(o *traceContextOptions) {
+		o.traceIDs = make([]string, len(traceIDs))
+		copy(o.traceIDs, traceIDs)
+	})
+}
+
+// TraceContext generates a TraceContext. Options may be provided to customize
+// the returned object.
+func TraceContext(options ...TraceContextOption) *flowpb.TraceContext {
+	opts := traceContextOptions{}
+	for _, opt := range options {
+		opt.apply(&opts)
+	}
+	traceID := fakeTraceID()
+	if len(opts.traceIDs) != 0 {
+		traceID = opts.traceIDs[rand.Intn(len(opts.traceIDs))]
+	}
+	return &flowpb.TraceContext{
+		Parent: &flowpb.TraceParent{
+			TraceId: traceID,
+		},
+	}
+}
+
+// fakeTraceID generates a fake trace ID. See the W3C Trace Context
+// specification for details:
+// https://www.w3.org/TR/trace-context/#trace-id
+func fakeTraceID() string {
```

Comment:
Any reason we don't just use an actual library to generate a traceID?

Reply:
I just thought it's not needed in this context (generating fake data). The trace ID is simply a 16-byte array where the only invalid value is all bytes as zero so something easy enough to generate. There are considerations to have when generating trace IDs for a "real" use-case (see [this document](https://www.w3.org/TR/trace-context/#considerations-for-trace-id-field-generation)) but I don't think it applies here.